simpleline/event_loop/main_loop.py

```diff
@@ -130,23 +155,38 @@ def _mainloop(self):
                 break
 
     def process_signals(self, return_after=None):
```

Comment:
I've added comments and also wrapped this functionality to methods. I think it should be much cleaner now.